src/tests/ftest/util/server_utils.py

```diff
@@ -48,6 +53,170 @@ class ServerFailed(Exception):
+            ServerFailed: if the specified job manager is unsupported.
+
+        Returns:
+            str: daos_sever launch command
+
+        """
+        if manager is None:
+            # Run locally
+            return self.__str__()
+
+        elif manager.endswith("orterun"):
+            assign_env = ["{}={}".format(key, val) for key, val in env.items()]
+            args = [
+                "-np {}".format(len(self.hosts)),
+                "-hostfile {}".format(self.hostfile),
+                "--enable-recovery",
+            ]
+            args.extend(["-x {}".format(assign) for assign in assign_env])
+            if uri is not None:
+                args.append("--report-uri {}".format(uri))
+
+        else:
+            raise ServerFailed("Unsupported job manager: {}".format(manager))
+
+        return "{} {} {}".format(manager, " ".join(args), self.__str__())
+
+    def start(self, manager, verbose=True, env=None, timeout=600):
+        """Start the daos server on each specified host.
+
+        Args:
+            manager (str): mpi job manager command
+            verbose (bool, optional): [description]. Defaults to True.
+            env ([type], optional): [description]. Defaults to None.
+            timeout (int, optional): [description]. Defaults to 600.
+
+        Raises:
+            ServerFailed: if there are issues starting the servers
+
+        """
+        if self.process is None:
+            # Start the daos server as a subprocess
+            kwargs = {
+                "cmd": self.get_launch_command(manager),
+                "verbose": verbose,
+                "allow_output_check": "combined",
+                "shell": True,
+                "env": env,
+                "sudo": True,
+            }
+            self.process = process.SubProcess(**kwargs)
+            self.process.start()
+
+            # Wait for 'DAOS I/O server' messages to appear in the daos_server
+            # output indicating that the servers have started
+            start_time = time.time()
+            start_msgs = 0
+            timed_out = False
+            while start_msgs != len(self.hosts) and not timed_out:
+                output = self.process.get_stdout()
+                start_msgs = len(re.findall("DAOS I/O server", output))
+                timed_out = time.time() - start_time > timeout
+
+            if start_msgs != len(self.hosts):
+                err_msg = "{} detected starting {}/{} servers".format(
+                    "Timed out" if timed_out else "Error",
+                    start_msgs, len(self.hosts))
+                print("{}:\n{}".format(err_msg, self.process.get_stdout()))
+                raise ServerFailed(err_msg)
+
+    def stop(self):
+        """Stop the process running the daos servers.
+
+        Raises:
+            ServerFailed: if there are errors stopping the servers
+
+        """
+        if self.process is not None:
+            signal_list = [9, None, None, None, None, 15]
```

Comment:
Thanks Brian, sorry for the mix-up Amanda.  How long should we wait for each signal to have an impact before sending the next, 1s, 2s, more?  Do we need both TERM and HUP? Should QUIT be called before KILL?  We have the ability to have different wait times between signals, e.g. send INT, wait 3s, if still running send TERM, wait 2s, if still running send QUIT, wait 1s, if still running send KILL:
`[2, None, None, None, 15, None, None, 3, None, 9]`

Reply:
So the answer to the question of how long to wait after a signal is sent is quite subjective to the process receiving it and how much work it wants to do as it winds down.  So how long is reasonable to wait is probably a question for some of the more knowledgeable DAOS devs, in respect to how much time to they want a DAOS server that has received a signal to have to wind-down.  Perhaps there is no signal catching (and wind down) in DAOS servers (yet) but the devs would know better than I.

Frankly, the INTR (^C) that `orterun` sends to the DAOS servers should be all that it takes to gracefully stop a DAOS server and let it wind down.  So the wait after that is probably the most important.  If the server is still running after that, it's most likely in a bad state any way and sending more signals won't push it along to wind down and then shut down.  At that point, just for good measure, you probably want TERM, HUP and then finally KILL.  And even that list is probably overkill.  I.e. if TERM doesn't stop it, neither will HUP and only KILL will kill it.

It would be well worthwhile getting guidance from the DAOS devs on all of this.  I suspect they will say let `orterun` send it's INTR, [ wait ... seconds ], then send a KILL if it's still there.  But let's see what they say.

And then keep in mind that all of this is subject to change when we are no longer starting DAOS servers with `orterun`.  :smile:

Reply:
I'll use the suggestion `[2, None, None, None, 15, None, None, 3, None, 9]` for now and ask in the dev forum in the mean time.

Reply:
Strictly speaking, `orterun` should send the SIGINT to all of the daos server processes itself.  I suppose it might not hurt to resend that as our first attempt at gracefully shutting it down.

As for using numbers, why not `import signal` and use the symbolic names like `signal.SIGINT`, etc.?

Reply:
It's a bit shorter with just the numbers, but I don't mind specifying the symbolic names so that it can be clear what signals are being sent.

Reply:
Using the names is both more portable as well as much more clear about which signals are being sent.